test/framework/machinehealthcheck_helpers.go

```diff
@@ -0,0 +1,194 @@
+	Expect(ctx).NotTo(BeNil(), "ctx is required for DiscoverMachineHealthChecksAndWaitForRemediation")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling DiscoverMachineHealthChecksAndWaitForRemediation")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling DiscoverMachineHealthChecksAndWaitForRemediation")
+	Expect(input.MachineDeployments).ToNot(BeEmpty(), "Invalid argument. input.MachineDeployments can't be empty when calling DiscoverMachineHealthChecksAndWaitForRemediation")
+
+	fmt.Fprintf(GinkgoWriter, "Discovering machine health check resources\n")
+	machineHealthChecks := GetMachineHealthChecksByCluster(ctx, GetMachineHealthChecksByClusterInput{
+		Lister:      input.ClusterProxy.GetClient(),
+		ClusterName: input.Cluster.Name,
+		Namespace:   input.Cluster.Namespace,
+	})
+	if len(machineHealthChecks) == 0 {
+		return
+	}
+
+	for _, mhc := range machineHealthChecks {
+		if len(mhc.Spec.UnhealthyConditions) < 1 {
+			continue
+		}
+		fmt.Fprintf(GinkgoWriter, "Patching Machines in MachineDeployments with MachineHealthCheck label selectors\n")
+		for _, md := range input.MachineDeployments {
+			Expect(len(mhc.Spec.Selector.MatchLabels)).To(BeNumerically(">", 0))
+
+			selectorMap, err := metav1.LabelSelectorAsMap(&md.Spec.Selector)
+			Expect(err).ToNot(HaveOccurred())
+
+			ms := &clusterv1.MachineSetList{}
+			Expect(input.ClusterProxy.GetClient().List(ctx, ms, client.InNamespace(input.Cluster.Namespace), client.MatchingLabels(selectorMap))).To(Succeed())
+
+			if len(ms.Items) == 0 {
+				continue
+			}
+			machineSet := ms.Items[0]
+			selectorMap, err = metav1.LabelSelectorAsMap(&machineSet.Spec.Selector)
+			Expect(err).ToNot(HaveOccurred())
+
+			machines := &clusterv1.MachineList{}
+			Expect(input.ClusterProxy.GetClient().List(ctx, machines, client.InNamespace(machineSet.Namespace), client.MatchingLabels(selectorMap))).To(Succeed())
+
+			for _, machine := range machines.Items {
+				machineCopy := machine.DeepCopy()
+				patchHelper, err := patch.NewHelper(machineCopy, input.ClusterProxy.GetClient())
+				Expect(err).ToNot(HaveOccurred())
+				machine.SetLabels(mhc.Spec.Selector.MatchLabels)
+				Expect(patchHelper.Patch(ctx, machineCopy)).To(Succeed())
+			}
+		}
+
+		fmt.Fprintf(GinkgoWriter, "Ensuring there is at least 1 Machine that MachineHealthCheck is matching\n")
+		machines := GetMachinesByMachineHealthCheck(context.TODO(), GetMachinesByMachineHealthCheckInput{
+			Lister:             input.ClusterProxy.GetClient(),
+			ClusterName:        input.Cluster.Name,
+			Namespace:          input.Cluster.Namespace,
+			MachineHealthCheck: mhc,
+		})
+		if len(machines) == 0 {
+			continue
+		}
+
+		fmt.Fprintf(GinkgoWriter, "Patching MachineHealthCheck unhealthy condition to one of the nodes\n")
```

Comment:
> Patching MachineHealthCheck unhealthy condition to one of the nodes

IMO This can be a new helper, SetNodeCondition

Reply:
It is already a helper, PatchNodeCondition.